sys/include/net/ng_netapi.h

```diff
@@ -0,0 +1,146 @@
+ *                    +--------------------------+
+ *                    |      Network layer       |
+ *                    +--------------------------+
+ *                       ^         ^         ^
+ *                     netapi    netapi    netapi
+ *                       v         v         v
+ *                   +-------+ +-------+ +-------+
+ *                   |  MAC  | |  MAC  | |  MAC  |
+ *                   +-------+ +-------+ +-------+
+ *                   | Radio | | Radio | | Radio |
+ *                   +-------+ +-------+ +-------+
+ *
+ * @{
+ *
+ * @file
+ * @brief       Generic interface to communicate with network modules
+ *
+ * @author      Martine Lenders <mlenders@inf.fu-berlin.de>
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef NG_NETAPI_H_
+#define NG_NETAPI_H_
+
+#include "kernel.h"
+#include "thread.h"
+#include "ng_netconf.h"
+#include "ng_pktsnip.h"
+
+
+/**
+ * @brief   Message type for passing data up the network stack
+ */
+#define NG_NETAPI_MSG_TYPE_RCV          (0x0201)
+
+/**
+ * @brief   Message type for passing data down the network stack
+ */
+#define NG_NETAPI_MSG_TYPE_SND          (0x0202)
+
+/**
+ * @brief   Message type for setting options of network modules
+ */
+#define NG_NETAPI_MSG_TYPE_SETOPT       (0x0203)
+
+/**
+ * @brief   Message type for getting options from network modules
+ */
+#define NG_NETAPI_MSG_TYPE_GETOPT       (0x0204)
+
+/**
+ * @brief   Message type for replying to get and set option messages
+ */
+#define NG_NETAPI_MSG_TYPE_ACK          (0x0205)
+
+/**
+ * @brief   Data structure to be send for setting and getting options
+ */
+typedef struct {
+    ng_netconf_opt_t opt;       /**< the option to get/set */
+    uint16_t context;           /**< (optional) context for that option */
+    void *data;                 /**< data to set or buffer to read into */
+    uint16_t data_len;          /**< size of the data / the buffer */
+} ng_netapi_opt_t;
+
+/**
+ * @brief   Shortcut function for sending *NETAPI_CMD_SND* messages
+ *
+ * @param[in] pid       PID of the targeted network module
+ * @param[in] pkt       pointer into the packet buffer holding the data to send
+ *
+ * @return              1 if packet was successfully delivered
+ * @return              -1 on error (invalid PID or no space in queue)
+ */
+int ng_netapi_send(kernel_pid_t pid, ng_pktsnip_t *pkt);
+
+/**
+ * @brief   Shortcut function for send *NETAPI_CMD_GETOPT* messages and parsing
+ *          the returned *NETAPI_CMD_ACK* message
+ *
+ * @param[in] pid       PID of the targeted network module
+ * @param[in] opt       option to get
+ * @param[in] context   (optional) context to the given option
+ * @param[in] data      pointer to buffer for reading the option's value
+ * @param[in] data_len  size of the given buffer
+ *
+ * @return              value returned by the ACK message
+ */
+int ng_netapi_get(kernel_pid_t pid, nt_netconf_opt_t opt, uint16_t context,
```

Comment:
nope. `ng_netapi_opt_t` is basically an 'internal' helper data structure for netapi. So hiding this by providing the function above makes the usage of netapi easier -> and thats what this shortcut function is for...